Jenkinsfile

```diff
@@ -50,6 +52,11 @@ matrices = [
     "RUNTIME": DEFAULT_RUNTIME,
     "CYTHON": DEFAULT_CYTHON
   ],
+  "HCD": [
+    "SERVER": DEFAULT_HCD,
+    "RUNTIME": DEFAULT_RUNTIME,
+    "CYTHON": DEFAULT_CYTHON
+  ],
```

Comment:
Discussed last week but the [default matrix](https://github.com/datastax/python-driver/blob/master/Jenkinsfile#L86) used for most test builds is the "SMOKE" matrix... and it seems to me we'd prefer to bring HCD into that matrix rather than introducing a new matrix for HCD (which would have to be explicitly triggered).  To that end this PR should:

* Remove the "HCD" matrix above
* Either add HCD versions to the DEFAULT_DSE array or keep them in the DEFAULT_HCD array
* Update the server selection logic below (or [here](https://github.com/datastax/python-driver/blob/master/Jenkinsfile#L54)) to include HCD versions in some way.

There is some subtlety here.  We're already pulling the most recent DSE version for smoke tests via the `DEFAULT_DSE.takeRight(1)` call.  What we probably want to wind up with is smoke tests running against the current DSE minor (6.9.x), the previous DSE minor (6.8.x) and HCD.  You can make this happen whether you keep a separate array for HCD or fold i into the DEFAULT_DSE array.

Keep in mind that every new server backend adds 2 x (number of Python versions) builds to the Jenkins runs... so we want to be judicious about what we add here.

Reply:
Very good catch. Fixed.